stl/inc/chrono

```diff
@@ -2086,6 +2091,720 @@ namespace chrono {
+        _NODISCARD auto operator()(const utc_time<_Duration>& _Time) const
+            noexcept(noexcept(_DestClock::from_utc(_Time))) /* strengthened */
+            requires _Convertible_from_utc_time<_DestClock, _Duration> {
+            return _DestClock::from_utc(_Time);
+        }
+    };
+
+    // [time.clock.cast.fn]
+
+    template <class _DestClock, class _SourceClock, class _Duration>
+    _NODISCARD auto clock_cast(const time_point<_SourceClock, _Duration>& _Time) {
+        constexpr bool _Has_direct_conversion =
+            is_invocable_v<clock_time_conversion<_DestClock, _SourceClock>, decltype(_Time)>;
+
+        constexpr bool _Utc_from_src  = _Convertible_to_utc_time<_SourceClock, _Duration>;
+        constexpr bool _Sys_from_src  = _Convertible_to_sys_time<_SourceClock, _Duration>;
+        constexpr bool _Dest_from_utc = _Convertible_from_utc_time<_DestClock, _Duration>;
+        constexpr bool _Dest_from_sys = _Convertible_from_sys_time<_DestClock, _Duration>;
+
+        constexpr bool _Has_utc_conversion = _Dest_from_utc && _Utc_from_src;
+        constexpr bool _Has_sys_conversion = _Dest_from_sys && _Sys_from_src;
+        static_assert(_Has_direct_conversion || !(_Has_utc_conversion && _Has_sys_conversion),
+            "A two-step clock time conversion is required to be unique, either through utc_clock or system_clock, but "
+            "not both [time.clock.cast.fn]/2.");
+
+        constexpr bool _Has_sys_utc_conversion = _Dest_from_sys && _Utc_from_src;
+        constexpr bool _Has_utc_sys_conversion = _Dest_from_utc && _Sys_from_src;
+        static_assert(_Has_direct_conversion || _Has_utc_conversion || _Has_sys_conversion
+                          || !(_Has_utc_sys_conversion && _Has_sys_utc_conversion),
+            "A three-step clock time conversion is required to be unique, either utc-to-system or system-to-utc, but "
+            "not both [time.clock.cast.fn]/2.");
+
+        if constexpr (_Has_direct_conversion) {
+            return clock_time_conversion<_DestClock, _SourceClock>{}(_Time);
+        } else if constexpr (_Has_utc_conversion) {
+            return clock_time_conversion<_DestClock, utc_clock>{}(
+                clock_time_conversion<utc_clock, _SourceClock>{}(_Time));
+        } else if constexpr (_Has_sys_conversion) {
+            return clock_time_conversion<_DestClock, system_clock>{}(
+                clock_time_conversion<system_clock, _SourceClock>{}(_Time));
+        } else if constexpr (_Has_sys_utc_conversion) {
+            return clock_time_conversion<_DestClock, system_clock>{}(clock_time_conversion<system_clock, utc_clock>{}(
+                clock_time_conversion<utc_clock, _SourceClock>{}(_Time)));
+        } else if constexpr (_Has_utc_sys_conversion) {
+            return clock_time_conversion<_DestClock, utc_clock>{}(clock_time_conversion<utc_clock, system_clock>{}(
+                clock_time_conversion<system_clock, _SourceClock>{}(_Time)));

```

Comment:
Will clang-format let us do some nicer formatting here?  Something like what is actually depicted in the Standard (each nested conversion on its own line) would be nice 😸